components/shared/Form/Login/LoginForm.tsx

```diff
@@ -0,0 +1,96 @@
+"use client";
+import Input from "~/components/ui/Input";
+import Button from "~/components/ui/Button";
+import { signIn, signOut } from "next-auth/react";
+import { useState } from "react";
+import Link from "next/link";
+
+interface LoginFormProps {
+  type: string;
+  title: string;
+}
+
+//TODO: Add error handling with Zod and useForm hook: https://ui.shadcn.com/docs/components/form
+
+const LoginForm: React.FC<LoginFormProps> = ({ type, title }) => {
+  const providers = ["Google", "GitHub"];
+  const isSignUp = type === "sign-up";
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [fullName, setFullName] = useState("");
+  const [error, setError] = useState("");
+  return (
+    <div className="mt-4 flex w-full flex-col justify-center">
+      {/* {error && <p className="text-myRed-500">{error}</p>} */}
+      <h1 className="text-display2 text-myWhite-100">{title}</h1>
+      <div className="flex flex-col gap-4">
+        {isSignUp && (
+          <Input
+            value={fullName}
+            onChange={(e) => setFullName(e.target.value)}
+            name="full name"
+            type="text"
+            placeholder="Enter your full name"
+          />
+        )}
+        <Input
+          value={email}
+          onChange={(e) => setEmail(e.target.value)}
+          name="email"
+          type="email"
+          placeholder="Enter your email address"
+        />
+        <Input
+          value={password}
+          onChange={(e) => setPassword(e.target.value)}
+          name="password"
+          type="password"
+          placeholder="Enter your password"
+        />
+        <Button
+          backgroundColor="bg-primary-500"
+          textColor="text-myBlack-900"
+          text={title}
+          onClick={(e) => {
+            e.preventDefault();
+            //save the result and see if there is an error, also redirect to the home page
+            signIn("credentials", {
+              email: email,
+              password: password,
+              fullName: fullName,
+            });
```

Comment:
While this is fine- we can get much more control over what's happening by instead returning the result and telling it not to redirect immediately.

```js
    onClick={async (e) => {
      e.preventDefault();
      //save the result and see if there is an error, also redirect to the home page
      const result = await signIn("credentials", {
        email: email,
        password: password,
        fullName: fullName,
        redirect: false,
      });
      console.log(result);
  ```

Below, shows the 'result' on top when the password I enter is incorrect when I try and login. The bottom result is a successful login. 

![image](https://github.com/dalbanhi/git-note/assets/4108484/bb7da70d-2977-4f82-a04f-e21538406a96)

We can use these values to show an error or toast on the page letting the user know their password failed or redirect them manually.

Something to consider if we want to avoid the default "ERROR" page from next-auth 👍